<launch>
  <!-- filereader -->
  <arg name="fpa" default="0.01" />
  <arg name="file_path" default="$(find innoviz_ros)/../../../examples/example_recordings/Lidar_1.invz4_4/" />
  <arg name="frame_id" default="base_link" />
  <arg name="ref0" default="invz_reflection_0" />
  <arg name="ref1" default="invz_reflection_1" />
  <arg name="ref2" default="invz_reflection_2" />
  <arg name="summation" default="invz_summation_reflection" />
  <arg name="invz_log_path" default="" />
  <arg name="invz_log_level" default="3" /> <!-- API log levels: 3=Warning 4=Error 5=Fatal 6=None -->
  <arg name="ros_log_level" default="1" />
  <arg name="config_file_path" default="$(find innoviz_ros)/../../../examples/lidar_configuration_files/recording_ignore_blooming_config.json"/>
  <arg name="pcl_timestamp_option" default="received"/>

  <!-- test -->
  <arg name="subscription_timeout_milliseconds" default="2000" />
  <arg name="message_timeout_milliseconds" default="2000" />

  <!-- filereader -->
  <node  pkg="innoviz_ros" type="Innoviz_one_FileReader" name="invz_publisher" output="screen">
    <param name="file_path" value="$(arg file_path)" />
    <param name="config_file_path" value="$(arg config_file_path)" />
    <param name="reflection_0" value="$(arg ref0)" />
	  <param name="reflection_1" value="$(arg ref1)" /> 
	  <param name="reflection_2" value="$(arg ref2)" /> 	
	  <param name="summation_reflection" value="$(arg summation)" /> 
    <param name="invz_log_path" value="$(arg invz_log_path)"/>
	  <param name="invz_log_level" type="int" value="$(arg invz_log_level)"/>
    <param name="ros_log_level" type="int" value="$(arg ros_log_level)"/>
    <param name="false_positive_alarm" type="double" value="$(arg fpa)"/>
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="pcl_timestamp_option" value="$(arg pcl_timestamp_option)" />
  </node>

  <!-- test -->
  <node  pkg="innoviz_ros" type="Innoviz_one_FileReader_test" name="invz_publisher_tester" output="screen">
    <param name="subscription_timeout_milliseconds" value="$(arg subscription_timeout_milliseconds)" />
    <param name="message_timeout_milliseconds" value="$(arg message_timeout_milliseconds)" />
  </node>
  
</launch>
